FraudProofs.md

```diff
@@ -0,0 +1,158 @@
+which user's can be taken through manual validation step-by-step.
+Such a process can serve as a manual means of black-listing oracle keys.
+
+#### Version 0 `false_oracle_attestation_claim`
+
+1. type: ??? (`false_oracle_attestation_claim_v0`)
+2. data:
+   * [`oracle_attestation_proof`:`oracle_attestation_proof`]
+   * [`string`:`fraud_claim`]
+
+This message consists of an `oracle_attestation_proof` showing that a specific oracle attested to
+a specific message and a `fraud_claim` explaining why this attestation is fraudulent.
+
+## Proof of Invalid Oracle Attestation
+
+The next simplest form of fraud is if an oracle attests using some outcome outside of its announced
+and committed to outcome set from its `oracle_announcement`.
+
+Unlike false attestations, this kind of fraud can be automatically validated by client software.
+
+#### Version 0 `invalid_oracle_attestation_proof`
+
+1. type: ??? (`invalid_oracle_attestation_proof_v0`)
+2. data:
+   * [`oracle_attestation_proof`: `oracle_attestation_proof`]
+
+This type simply consists of an `oracle_attestation_proof` proving that a specific oracle attested to
+a specific `oracle_outcome` where this outcome is not allowed by that oracle's `oracle_announcement`.
+
+An important thing to keep in mind is that in the case of invalid attestations, there are no concerns related
+to recovering the `oracle_outcome` from on-chain information because there are no CETs or adaptor
+signatures built for such an outcome.
+In other words, this kind of fraud is only detectable and only important if the `oracle_outcome` is known
+as otherwise the invalid attestation is essentially an encrypted message and such an attestation is just
+a random 32 bytes that cannot be proven to even be an attestation at all.
+
+## Proof of Oracle Equivocation
+
+Another form of oracle fraud for which there exists a straightforward automatic validation is equivocation.
+Equivocation is when an oracle attests to multiple messages for the same event.
+
+Validation in this case simply requires validating that both attestations exist using `oracle_attestation_proof`s.
+
+#### Version 0 `oracle_equivocation_proof`
+
+1. type: ??? (`oracle_equivocation_proof_v0`)
+2. data:
+   * [`oracle_attestation_proof`:`oracle_attestation_proof_1`]
+   * [`oracle_attestation_proof`:`oracle_attestation_proof_2`]
+
+The most general kind of oracle equivocation proof is simply the combination of two `oracle_attestation_proof`s
+for the same `oracle_announcement` but for different `oracle_outcome`s.
+
+#### Version 1 `oracle_equivocation_proof`
+
+1. type: ??? (oracle_equivocation_proof_v1)
+2. data:
+   * [`oracle_announcement`:`oracle_announcement`]
+   * [`nb_signatures*string`:`outcomes_1`]
+   * [`32*bytes`:`oracle_attestation_1`]
+   * [`nb_signatures*string`:`outcomes_2`]
+   * [`32*bytes`:`oracle_attestation_2`]
+   * [`32*bytes`:`oracle_private_key`]
```

Comment:
Is it necessary to put this in since anyway it can be recomputed?

Reply:
it is not, and I'm open to removing it just thought it might be fun :P

Reply:
Sorry if I'm being boring but I'd rather not have it.